elements/lisk-p2p/src/peer_book/base_list.ts

```diff
@@ -13,146 +13,195 @@
  *
  */
 import { ExistingPeerError } from '../errors';
-import { P2PPeerInfo } from '../p2p_types';
-import { evictPeerRandomlyFromBucket, getBucketId, PEER_TYPE } from '../utils';
+import { P2PEnhancedPeerInfo, P2PPeerInfo } from '../p2p_types';
+import {
+	evictPeerRandomlyFromBucket,
+	getBucketId,
+	PEER_TYPE,
+	sanitizeEnhancedPeerInfo,
+} from '../utils';
 
 export interface PeerListConfig {
-	readonly peerBucketCount: number;
-	readonly peerBucketSize: number;
+	readonly numOfBuckets: number;
+	readonly bucketSize: number;
 	readonly secret: number;
 	readonly peerType: PEER_TYPE;
 }
 
-export interface CustomPeerInfo {
-	readonly peerInfo: P2PPeerInfo;
-	readonly dateAdded: Date;
+export type Bucket = Map<string, P2PEnhancedPeerInfo>;
```

Comment:
`PeersBucket`?

Reply:
Prefer `bucket` as `Peers` part would be redundant in a `p2p` bucket.